File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.cpp
class Solution {
    bool isEven(int& x) {
        return (x & 1) == 0;
    }
public:
    vector<int> sumEvenAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        vector<int> ans;
        
        int evenSum = 0;
        for(int& x: nums) {
            if(isEven(x))
                evenSum += x;
        }
        
        for(vector<int>& query: queries) {
            int& val = query[0];
            int& index = query[1];
            
            int initialValue = nums[index];
            nums[index] = initialValue + val;
            int updatedValue = nums[index];
            
            if(isEven(initialValue)) {
                evenSum -= initialValue;
            }
            if(isEven(updatedValue)) {
                evenSum += updatedValue;
            }
            
            
            ans.push_back(evenSum);
        }
        return ans;
   }
};

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        ans = []
        
        evenSum = 0
        
        for x in nums:
            if (x & 1) == 0:
                evenSum += x
        
        for query in queries:
            val, index = query
            
            initialValue = nums[index]
            nums[index] += val
            updatedValue = nums[index]
            
            if (initialValue & 1) == 0:
                evenSum -= initialValue
            
            if (updatedValue & 1) == 0:
                evenSum += updatedValue
            
            ans.append(evenSum)
            
        return ans